### PR TITLE
Change DEVELOPING.md to better reflect ported LLDB

### PR DESCRIPTION
**Pull request by mbrla0**
This PR changes `DEVELOPING.md` to better reflect Pwndbg after the LLDB port. It updates some code snippets, adds a few new notes, and explains the debugger-agnostic interfaces.

This is intended to help with https://github.com/pwndbg/pwndbg/issues/2522.

### CODE REVIEW

File: DEVELOPING.md [hunk 1]
```diff
@@ -135,19 +135,19 @@ def my_command(arg: str) -> None:
...
 ```
 
-In addition, you need to import this file in the `load_commands` function in `pwndbg/commands/__init__.py`. After this, running `my_command foo` in GDB will print out "Argument is foo".
+In addition, you need to import this file in the `load_commands` function in `pwndbg/commands/__init__.py`. After this, running `my_command foo` in GDB or LLDB will print out "Argument is foo".
 
 # Adding a Configuration Option
 
 ```python
-import pwndbg.gdblib.config
+import pwndbg.config
```
Comment (disconnect3d) [inline DEVELOPING.md]: Is this correct?
```
In [1]: import pwndbg.config
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], line 1
----> 1 import pwndbg.config

ModuleNotFoundError: No module named 'pwndbg.config'
```

Comment (mbrla0) [reply DEVELOPING.md]: Oh, right, it should be just `import pwndbg`.

File: DEVELOPING.md [hunk 2]
```diff
@@ -175,33 +175,103 @@ Feel free to update the list below!
...
+* Pwndbg has its own event system, and thanks to it we can set up code to be invoked in response to them. The event types and the conditions in which they occurr are defined and documented in the `EventType` enum, and functions are registered to be called on events with the `@pwndbg.dbg.event_handler` decorator. Both the enum and the decorator are documented in `pwndbg/dbg/__init__.py`.
 
 * We have a caching mechanism (["memoization"](https://en.wikipedia.org/wiki/Memoization)) which we use through Python's decorators - those are defined in `pwndbg/lib/cache.py` - just check its usages
 
 * To block a function before the first prompt was displayed use the `pwndbg.decorators.only_after_first_prompt` decorator.
 
-* Memory accesses should be done through `pwndbg/memory.py` functions
+* Memory accesses should be done through `pwndbg/aglib/memory.py` functions.
+
+* Process properties can be retrieved thx to `pwndbg/aglib/proc.py` - e.g. using `pwndbg.aglib.proc.pid` will give us current process pid
```
Comment (fidgetingbits) [inline DEVELOPING.md]: thx -> thanks

File: DEVELOPING.md [hunk 3]
```diff
@@ -175,33 +175,103 @@ Feel free to update the list below!
...
+supported debugger.
+
+As a matter of clarity, it makes sense to think of the Debugger API as a debugger-agnostic version
+of the `lldb` and `gdb` Python modules. Compared to both modules, it is much closer in spirit to
+`lldb` than to `gdb`.
+
+It is important to note that a lot of care must be exercised when adding things to the Debugger API,
+as one must always add implementations for all supported debuggers of whatever new functionality is
+being added, even if only to properly gate off debuggers in which the functionality is not supported.
+Aditionally, it is important to keep the Debugger API interfaces as terse as possible in order to
```
Comment (fidgetingbits) [inline DEVELOPING.md]: Additionally

File: DEVELOPING.md [hunk 4]
```diff
@@ -175,33 +175,103 @@ Feel free to update the list below!
...
+Here are some things one may want to do, along with how they can be achieved in the LLDB, GDB, and
+Pwndbg Debugger APIs.
 
+| Action | GDB/ | LLDB | Debugger API[^1] |
+| ------ | --- | ---- | ------------ |
+| Setting a breakpoint at an address | `gdb.Breakpoint("*<address>")` | `lldb.target.BreakpointCreateByAddress(<address>)` | `inf.break_at(BreakpointLocation(<address>))` |
+| Querying for the address of a symbol | `int(gdb.lookup_symbol(<name>).value().address)` | `lldb.target.FindSymbols(<name>).GetContextAtIndex(0).symbol.GetStartAddress().GetLoadAddress(lldb.target)` | `inf.symbol_address_from_name(<name>)` |
+| Setting a watchpoint at an address | `gdb.Breakpoint(f"(char[{<size>}])*{<address>}", gdb.BP_WATCHPOINT)` | `lldb.target.WatchAddress(<address>, <size>, ...)` | `inf.break_at(WatchpointLocation(<address>, <size>))` |
 
+[^1]: Many functions in the Debugger API are acessed through a `Process` object, which is usually
```
Comment (fidgetingbits) [inline DEVELOPING.md]: accessed

**Review by fidgetingbits — CHANGES_REQUESTED** [review]
Thanks for documenting all that!

**Comment by patryk4815** [comment]
image

Can you fix, 'Writing Tests' section (a lot of gdblib)

**Pull request merged**